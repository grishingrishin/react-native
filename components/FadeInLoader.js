import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Animated } from 'react-native';

class FadeInLoader extends Component {
  constructor() {
    super();

    this.state = {
      opacity: new Animated.Value(0),
    }
  }

  componentDidMount() {
    return Animated.timing(this.state.opacity, {
      toValue: 1,
      duration: 1000,
      useNativeDriver: true,
    }).start();
  }

  render() {
    return (
      <Animated.View style={{ opacity: this.state.opacity }}>
        {this.props.children}
      </Animated.View>
    )
  }
}

FadeInLoader.propTypes = {
  children: PropTypes.node.isRequired
}

export default FadeInLoader;