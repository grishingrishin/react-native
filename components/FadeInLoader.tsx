import React, { Component } from 'react';
import { Animated } from 'react-native';

interface FadeInLoaderProps {
  children: any
}

interface FadeInLoaderState {
  opacity: any
}

class FadeInLoader extends Component<FadeInLoaderProps, FadeInLoaderState> {
  constructor(props: FadeInLoaderProps) {
    super(props);

    this.state = {
      opacity: new Animated.Value(0),
    }
  }

  componentDidMount() {
    return Animated.timing(this.state.opacity, {
      toValue: 1,
      duration: 1000,
      useNativeDriver: true,
    }).start();
  }

  render() {
    return (
      <Animated.View style={{ opacity: this.state.opacity }}>
        {this.props.children}
      </Animated.View>
    )
  }
}

export default FadeInLoader;