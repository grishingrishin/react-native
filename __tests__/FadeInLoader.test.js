import React from 'react';
import { create } from 'react-test-renderer';
import FadeInLoader from '../components/FadeInLoader';
import { Text } from 'react-native';

// Disable warning: `useNativeDriver` is not supported because the native animated module is missing. Falling back to JS-based animation. 
// To resolve this, add `RCTAnimation` module to this app, or remove `useNativeDriver.
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
// Fix referenceError: You are trying to `import` a file after the Jest environment has been torn down.
jest.useFakeTimers();

describe('<FadeInLoader />', () => {
  describe('Jest tests', () => {
    it('renders correctly', () => {
      const tree = create(
        <FadeInLoader>
          <Text>Some test node</Text>
        </FadeInLoader>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
