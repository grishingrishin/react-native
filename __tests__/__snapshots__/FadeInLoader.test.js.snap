// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FadeInLoader /> Jest tests renders correctly 1`] = `
<View
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <Text>
    Some test node
  </Text>
</View>
`;
